src/NuGet.Core/NuGet.Build.Tasks/NuGet.Build.Tasks.csproj

```diff
@@ -50,6 +50,7 @@
     <PackageReference Include="Microsoft.Build.Utilities.Core" ExcludeAssets="runtime" />
     <!-- System.Security.Cryptography.Xml is a dependency of Microsoft.Build.Tasks.Core. Therefore, we have assigned the same ExcludeAssets value. -->
     <PackageReference Include="System.Security.Cryptography.Xml" ExcludeAssets="runtime" />
+    <PackageReference Include="System.Security.Cryptography.Pkcs" ExcludeAssets="runtime" />
```

Comment:
Actually, I don't think that excluding runtime is correct. Thoughts @jeffkl @nkolev92 ?

Since this is a direct package reference, it will override the transitive package reference from nuget.packaging, and therefore the dll will be excluded. I'm actually surprised tests don't fail at runtime because of this missing dll. Since this package reference is only being added for .NET Core, the biggest risk is if NuGet is the only component causing this package to be included in the .NET SDK, then if we merge this PR, when we insert the pkcs dll will be excluded from the SDK and then NuGet stops working. The way our `dotnet.integration.tests` patch the SDK for testing, it won't catch this type of error.

So, I think leave out the `ExcludeAssets` is the correct solution (honestly, all this would be simpler if transitive packages were only runtime by default, no buildTransitive, no compile). But most of our code has no business calling cryptography APIs, other than NuGet.Packaging, so `ExcludeAssets="compile"` wouldn't be terrible.

Reply:
I saw all other package references are already using `ExcludeAssets="runtime"` then followed the convention. Of course I can change to `ExcludeAssets="compile"`. Before doing that I'll wait feedback from @jeffkl @nkolev92

Reply:
I agree. I don't see a reason why `Exclude="Runtime"` is added here. 
I'm guessing the tests are passing because the SDK carries versions of this assembly. 

What's the reasoning behind adding the `Exclude="Runtime"` here @erdembayar

Reply:
> What's the reasoning behind adding the Exclude="Runtime" here @erdembayar
@nkolev92 

This convention was here before my change, according to blame it looks @kartheekp-ms  added line 51-52, he might know the reasoning.
https://github.com/NuGet/NuGet.Client/blob/b3c4e8094bbb924408180d700ae5b3d37713587c/src/NuGet.Core/NuGet.Build.Tasks/NuGet.Build.Tasks.csproj#L49-L52

Reply:
In addition to above it looks like line 49-50 was added by @donnie-msft in https://github.com/NuGet/NuGet.Client/pull/4761

Reply:
I don't have to remember, because it's in the first sentence of the PR you linked:

>  MSBuild does not support runtime scenarios where the Microsoft.Build DLL is placed in an output directory.

Reply:
> This convention was here before my change, according to blame it looks @kartheekp-ms added line 51-52, he might know the reasoning.

It's not a convention. The MSBuild packages are special, and System.Security.Cryptography.xml seems to be a result of that as well.

Reply:
> > This convention was here before my change, according to blame it looks @kartheekp-ms added line 51-52, he might know the reasoning.
> 
> It's not a convention. The MSBuild packages are special, and System.Security.Cryptography.xml seems to be a result of that as well.

I don't know much about MSBuild special packages. Do you want me to ask `Sourcebuild` or `dotnet/msbuild folks` to take a look at this?

Reply:
@nkolev92 
What makes `System.Security.Cryptography.Pkcs` different from ` System.Security.Cryptography.xml` already here? Both of them are transitive packages here. 
It looks @zivkan  and @jeffkl  approved the original https://github.com/NuGet/NuGet.Client/pull/4761, do you have any knowledge about this?

Reply:
The MSBuild packages are special, because NuGet.Build.Tasks[.*] are class libraries that implement MSBuild's Task (ITask?) interface. Therefore, the only way to run these libraries is to invoke MSbuild and have a `<UsingTask` element call into our library. But since msbuild.exe (or dotnet.exe) is the process dynamically loading our assembly, Microsoft.Build.*.dll is already loaded by MSBuild. Therefore, we don't have a need to ship those dlls with our task.

When I look at `{vs install directory}\MSBuild` and search for `System.Security.Cryptography.*.dll`, I don't see the pkcs dll. I also don't see the .xml assembly either, but is S.S.C.X a transitive dependency of NuGet.Packaging? My guess is that it's not.

However, this is about .NET (Core) TFMs, not .NET Framework, so looking at MSBuild's VS install isn't really representative.

If you really want to validate, you can build your branch locally, including packing the nupkgs (which build.ps1 does by default), the clone dotnet/sdk, add NuGet.Client's nupkgs directory as a package source, update the nuget.client package version to your local nupkg version number in the props file (eng\Package.Versions.props?), then build the SDK locally, and check if the .pkcs dll still makes it into the built SDK output directory.  But I think just removing the `ExcludeAssets="compile"` for safety would be quicker, as well as more technically correct of our package graph.

Reply:
Ok. I changed to `ExcludeAssets="compile"`